lib/presto-client/index.js

```diff
@@ -89,13 +99,19 @@ Client.prototype.request = function(opts, callback) {
 
     opts.headers[client.headers.USER_AGENT] = this.userAgent;
 
-    /**
-     * Apply an Authorization header if the user
-     * has specified a client.basic_auth object.
-     * client.basic_auth = { user: "<user>", password : "<password>"}
-     */
-    if (client.basic_auth)
+    if (client.authorization) {
+        opts.headers[client.headers.AUTHORIZATION] = authorization;
+    } else if (client.basic_auth) {
```

Comment:
I added this `else-if` clause just in case if anyone is modifying `basic_auth` property on `Client` instance, avoid making any breaking changes.

```js
// Example
const client = new Client(/* settings */);

// Modifies basic_auth after instantiation
client.basic_auth = { /* new user & password */ }

client.execute(/* ... */) // Will use the new basic_auth
```

Personally I do not like this backward compatibility support because I had to duplicate auth string building logic here.  @tagomoris  should we keep it or drop it to be cleaner?

Reply:
I feel that you should be a great professional programmer because of such consideration!
But I agree with your preference - we can avoid providing such compatibility. We have no documents/examples to show the way to set `basic_auth` lazily.
+1 to drop it to be clearer!

Reply:
Thank you, for the compliment, for this great library you have build, and for the effort you have put in maintaining this library 🤗

Backward compatibility code is removed now.